File: cash/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int const duli = 25;
int const pulu = 10;
int const lima = 5;
int const satu = 1;

int hasil_bagi;
int jumlah_koin;
int sisa;

void kembalian(int change);

int main(void)
{
    int c;
    do
    {
        c = get_int("Change owed: ");
    }
    while (c <= 0);
    kembalian(c);
}

void kembalian(int change)
{

    hasil_bagi = change / duli;
    jumlah_koin = hasil_bagi;
    sisa = change % duli;

    if (sisa != 0)
    {
        hasil_bagi = sisa / pulu;
        jumlah_koin = jumlah_koin + hasil_bagi;
        sisa = sisa % pulu;
    }

    if (sisa != 0)
    {
        hasil_bagi = sisa / lima;
        jumlah_koin = jumlah_koin + hasil_bagi;
        sisa = sisa % lima;
    }

    if (sisa != 0)
    {
        hasil_bagi = sisa / satu;
        jumlah_koin = jumlah_koin + hasil_bagi;
        sisa = sisa % satu;
    }
    printf("%i\n", jumlah_koin);
}

File: sentimental-cash/cash.py
from cs50 import get_float


def main():
    kembalian = 0
    while True:
        if kembalian > 0:
            break
        kembalian = get_float("Change owed: ")
    calc_kembalian(kembalian)


def calc_kembalian(kembali):
    DULI = 25
    PULU = 10
    LIMA = 5
    SATU = 1

    kembali = kembali * 100
    hasil_bagi = 0
    jumlah_koin = 0
    sisa = 0

    hasil_bagi = kembali // DULI
    jumlah_koin = hasil_bagi
    sisa = kembali % DULI

    if sisa != 0:
        hasil_bagi = sisa // PULU
        jumlah_koin += hasil_bagi
        sisa = sisa % PULU

    if sisa != 0:
        hasil_bagi = sisa // LIMA
        jumlah_koin += hasil_bagi
        sisa = sisa % LIMA

    if sisa != 0:
        hasil_bagi = sisa // SATU
        jumlah_koin += hasil_bagi
        sisa = sisa % SATU

    print(jumlah_koin)


main()
